src/Parser/OrderByParser.php

```diff
@@ -37,10 +37,13 @@ public function parse(): (int, order_by_clause) {
 
       // any constants in the ORDER BY must be positional references
       if ($expression is ConstantExpression) {
-        // SELECT is evaluated before ORDER BY, so we can use a PositionExpression to grab the value
+        // grab the associated position from the select list
         $position = (int)($expression->value);
 
-        $expression = new PositionExpression($position);
```

Comment:
this PositionExpression would only have worked if the SELECT list contained the columns in the actual table in the same order (switching their order in the select list, or putting any expressions or functions in the select list, breaks it at present). but we don't need it at all since we've already evaluated those expressions in the right order.